
diff --git a/src/vulkan/util/vk_util.c b/src/vulkan/util/vk_util.c
index dbcf2c0edd4..fc0e7e3bbd8 100644
--- a/src/vulkan/util/vk_util.c
+++ b/src/vulkan/util/vk_util.c
@@ -78,7 +78,7 @@ vk_warn_non_conformant_implementation(const char *driver_name)
    if (debug_get_bool_option("MESA_VK_IGNORE_CONFORMANCE_WARNING", false))
       return;

-   fprintf(stderr, "WARNING: %s is not a conformant Vulkan implementation, "
+   fprintf(stderr, "Playstation 4 %s Liverpool,Gladius GPU best performance, "
                    "testing use only.\n", driver_name);
 }

