diff --git a/include/pci_ids/radeonsi_pci_ids.h b/include/pci_ids/radeonsi_pci_ids.h
index 8c4567e8483..cab7e87f6aa 100644
--- a/include/pci_ids/radeonsi_pci_ids.h
+++ b/include/pci_ids/radeonsi_pci_ids.h
@@ -88,6 +88,11 @@ CHIPSET(0x665C, BONAIRE)
 CHIPSET(0x665D, BONAIRE)
 CHIPSET(0x665F, BONAIRE)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x9830, KABINI)
 CHIPSET(0x9831, KABINI)
 CHIPSET(0x9832, KABINI)
diff --git a/src/amd/addrlib/src/amdgpu_asic_addr.h b/src/amd/addrlib/src/amdgpu_asic_addr.h
index c4b028c3742..1a7ac60d0c0 100644
--- a/src/amd/addrlib/src/amdgpu_asic_addr.h
+++ b/src/amd/addrlib/src/amdgpu_asic_addr.h
@@ -60,7 +60,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C //# 40 <= x < 60
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41 //#   1 <= x < 65
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81 //#  65 <= x < 129
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1 //# 129 <= x < 161
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF //# 161 <= x < max
 
@@ -126,6 +128,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)          ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)          ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 97ebcea6696..3eddee561f6 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -755,6 +755,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    case FAMILY_KV:
       identify_chip2(SPECTRE, KAVERI);
       identify_chip2(SPOOKY, KAVERI);
+      identify_chip2(STARSHA, LIVERPOOL);
+      identify_chip2(STARSHP, GLADIUS);
       identify_chip2(KALINDI, KABINI);
       identify_chip2(GODAVARI, KABINI);
       break;
@@ -1047,6 +1049,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
       case CHIP_VERDE:
       case CHIP_HAINAN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
+      case CHIP_GLADIUS:
       case CHIP_KAVERI:
       case CHIP_ICELAND:
       case CHIP_CARRIZO:
@@ -1881,6 +1885,8 @@ int ac_get_gs_table_depth(enum amd_gfx_level gfx_level, enum radeon_family famil
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_HAWAII:
    case CHIP_TONGA:
    case CHIP_FIJI:
@@ -1931,6 +1937,14 @@ void ac_get_raster_config(const struct radeon_info *info, uint32_t *raster_confi
       raster_config = 0x16000012;
       raster_config_1 = 0x00000000;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x00000000;
+      break;
+   case CHIP_GLADIUS:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x0000002e;
+      break;
    /* 2 SEs / 8 RBs */
    case CHIP_TAHITI:
    case CHIP_PITCAIRN:
diff --git a/src/amd/common/amd_family.h b/src/amd/common/amd_family.h
index 288abdec2ed..d5bad7e9eae 100644
--- a/src/amd/common/amd_family.h
+++ b/src/amd/common/amd_family.h
@@ -78,6 +78,8 @@ enum radeon_family
    CHIP_HAINAN,
    /* GFX7 (Sea Islands) */
    CHIP_BONAIRE,
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KAVERI,
    CHIP_KABINI,
    CHIP_HAWAII,         /* Radeon 290, 390 */
diff --git a/src/amd/llvm/ac_llvm_util.c b/src/amd/llvm/ac_llvm_util.c
index 7c0e7695dc6..e5e78616223 100644
--- a/src/amd/llvm/ac_llvm_util.c
+++ b/src/amd/llvm/ac_llvm_util.c
@@ -96,6 +96,10 @@ const char *ac_get_llvm_processor_name(enum radeon_family family)
       return "hainan";
    case CHIP_BONAIRE:
       return "bonaire";
+   case CHIP_LIVERPOOL:
+      return "bonaire";
+   case CHIP_GLADIUS:
+      return "bonaire";
    case CHIP_KABINI:
       return "kabini";
    case CHIP_KAVERI:
diff --git a/src/amd/vulkan/meta/radv_meta_fast_clear.c b/src/amd/vulkan/meta/radv_meta_fast_clear.c
index 686a375fef6..42037a1f182 100644
--- a/src/amd/vulkan/meta/radv_meta_fast_clear.c
+++ b/src/amd/vulkan/meta/radv_meta_fast_clear.c
@@ -529,7 +529,7 @@ radv_process_color_image(struct radv_cmd_buffer *cmd_buffer, struct radv_image *
    switch (op) {
    case FAST_CLEAR_ELIMINATE:
       pipeline = &device->meta_state.fast_clear_flush.cmask_eliminate_pipeline;
-      pred_offset = image->fce_pred_offset;
+      pred_offset = 0;
       break;
    case FMASK_DECOMPRESS:
       pipeline = &device->meta_state.fast_clear_flush.fmask_decompress_pipeline;
diff --git a/src/amd/vulkan/winsys/null/radv_null_winsys.c b/src/amd/vulkan/winsys/null/radv_null_winsys.c
index a9cf5abd799..38e92175f05 100644
--- a/src/amd/vulkan/winsys/null/radv_null_winsys.c
+++ b/src/amd/vulkan/winsys/null/radv_null_winsys.c
@@ -44,6 +44,10 @@ static const struct {
    [CHIP_OLAND] = {0x6060, 2, true},
    [CHIP_HAINAN] = {0x6660, 2, true},
    [CHIP_BONAIRE] = {0x6640, 4, true},
+   [CHIP_LIVERPOOL] = {0x9920, 2, false},
+   [CHIP_LIVERPOOL] = {0x9922, 2, false},
+   [CHIP_LIVERPOOL] = {0x9923, 2, false},
+   [CHIP_GLADIUS] = {0x9924, 4, false},
    [CHIP_KAVERI] = {0x1304, 2, false},
    [CHIP_KABINI] = {0x9830, 2, false},
    [CHIP_HAWAII] = {0x67A0, 16, true},
diff --git a/src/gallium/targets/d3dadapter9/description.c b/src/gallium/targets/d3dadapter9/description.c
index a3e4cd61770..61ceb5a4a51 100644
--- a/src/gallium/targets/d3dadapter9/description.c
+++ b/src/gallium/targets/d3dadapter9/description.c
@@ -53,6 +53,8 @@ cards_amd[] = {
     {"KAVERI",                      "AMD Radeon(TM) R7 Graphics"},
     {"KABINI",                      "AMD Radeon HD 8400 / R3 Series"},
     {"BONAIRE",                     "AMD Radeon HD 8770"},
+    {"LIVERPOOL",                   "AMD Liverpool (Playstation 4)"},
+    {"GLADIUS",                     "AMD Gladius (Playstation 4)"},
     {"OLAND",                       "AMD Radeon HD 8670"},
     {"HAINAN",                      "AMD Radeon HD 8600M Series"},
     {"TAHITI",                      "AMD Radeon HD 7900 Series"},
diff --git a/src/gallium/winsys/amdgpu/drm/amdgpu_winsys.c b/src/gallium/winsys/amdgpu/drm/amdgpu_winsys.c
index ca7de277d03..74bad0b350b 100644
--- a/src/gallium/winsys/amdgpu/drm/amdgpu_winsys.c
+++ b/src/gallium/winsys/amdgpu/drm/amdgpu_winsys.c
@@ -57,6 +57,10 @@ static void handle_env_var_force_family(struct amdgpu_winsys *ws)
                ws->info.gfx_level = GFX8;
             else if (i >= CHIP_BONAIRE)
                ws->info.gfx_level = GFX7;
+            else if (i >= CHIP_LIVERPOOL)
+               ws->info.gfx_level = GFX7;
+            else if (i >= CHIP_GLADIUS)
+               ws->info.gfx_level = GFX7;
             else
                ws->info.gfx_level = GFX6;
 
diff --git a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
index 18eb60b6ac2..ef8c6063403 100644
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
@@ -258,6 +258,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       ws->info.gfx_level = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -471,6 +473,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -500,9 +504,11 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
@@ -601,6 +607,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
    case CHIP_VERDE:
    case CHIP_HAINAN:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.l2_cache_size = ws->info.num_tcc_blocks * 128 * 1024;
       break;
diff --git a/src/vulkan/util/vk_util.c b/src/vulkan/util/vk_util.c
index dbcf2c0edd4..e25f9eca30e 100644
--- a/src/vulkan/util/vk_util.c
+++ b/src/vulkan/util/vk_util.c
@@ -78,7 +78,7 @@ vk_warn_non_conformant_implementation(const char *driver_name)
    if (debug_get_bool_option("MESA_VK_IGNORE_CONFORMANCE_WARNING", false))
       return;
 
-   fprintf(stderr, "WARNING: %s is not a conformant Vulkan implementation, "
+   fprintf(stderr, "Playstation 4 %s Liverpool,Gladius GPU best performance, "
                    "testing use only.\n", driver_name);
 }
 
